sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -37,6 +37,19 @@ import org.apache.spark.sql.sources.{ResolvedDataSource, CreateTableUsingAsSelec
 final class DataFrameWriter private[sql](df: DataFrame) {
 
   /**
+   * Check if the provided dataframe has multiple columns
+   * before writing it to an external source
+   */
+  private val error : Unit = {
+    if (df.schema.fieldNames.length != df.schema.fieldNames.distinct.length) {
```

Comment:
the best place to do this might be looking at a logical plan that contains an output operator, rather than putting it in the writer itself.